litr/src/main/java/com/linkedin/android/litr/thumbnails/VideoThumbnailExtractor.kt

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2021 LinkedIn Corporation
+ * All Rights Reserved.
+ *
+ * Licensed under the BSD 2-Clause License (the "License").  See License in the project root for
+ * license information.
+ */
+package com.linkedin.android.litr.thumbnails
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import com.linkedin.android.litr.ExperimentalThumbnailsApi
+import com.linkedin.android.litr.filter.GlFilter
+import com.linkedin.android.litr.render.GlThumbnailRenderer
+import com.linkedin.android.litr.render.ThumbnailRenderer
+import com.linkedin.android.litr.thumbnails.behaviors.ExtractionBehavior
+import com.linkedin.android.litr.thumbnails.behaviors.MediaMetadataExtractionBehavior
+import com.linkedin.android.litr.thumbnails.queue.ComparableFutureTask
+import com.linkedin.android.litr.thumbnails.queue.PriorityExecutorUtil
+import java.util.concurrent.*
+
+/**
+ * Provides the entry point for thumbnail extraction.
+ *
+ * This class uses a single, dedicated thread to schedule jobs. The priority of each job can be specified within [ThumbnailExtractParameters].
+ *
+ * @param context The application context.
+ * @param listenerLooper The looper on which [extract] listener events will be processed.
+ * @param extractionBehavior The behavior to use for extracting frames from media.
+ */
+@ExperimentalThumbnailsApi
+class VideoThumbnailExtractor @JvmOverloads constructor(
```

Comment:
Can we rename this (and related stack) to `VideoFrameExtractor`? This should be a general purpose utility, thumbnails being one of use cases.